src/urllib3/response.py

```diff
@@ -986,6 +986,9 @@ def read(
 
         if amt is None:
             data = self._decode(data, decode_content, flush_decoder)
+            if len(self._decoded_buffer) != 0:
+                data = self._decoded_buffer.get_all() + data
+
             if cache_content:
```

Comment:
Perhaps we need to set `cache_content = False` when `read(amt=None, cache_content=True)` follows a partial read